### DISCUSSION TITLE
WPF ListBox设置SelectIndex=0，并自动触发SelectionChanged事件

### DISCUSSION THREAD

**Discussion by SnowO1127** [BYJRK/DotNet-Discussions#250] [WPF 开发]
使用HandyControl默认SelectIndex=0，默认选中第一个，并没有高亮，也不会自动触发SelectionChanged事件

**Comment by BYJRK** [comment]
你想要实现的效果是什么？

**Reply by SnowO1127** [reply]
ListBox默认选中第一个，并高亮，触发SelectionChange事件

**Comment by lmccccl** [comment]
贴一下代码看看

**Comment by hinadesu39** [comment]
试试在ListBox的Loaded事件中去初始化SelectIndex

**Comment by SlimeNull** [comment]
我在 UWP 中遇到过类似的问题，最终解决方案是，延时1ms。虽然很扯但确实是这样解决的。你可以试试。（但注意，不是 Thread.Sleep，这玩意儿会阻塞线程，我是想表示，在窗口加载完毕后1ms，再去初始化。因为某些东西的布局，ApplyTemplate 之类的东西，调用时机会有问题。
